crates/chainspec/src/api.rs

```diff
@@ -13,6 +15,9 @@ pub trait EthChainSpec: Send + Sync + Unpin + Debug + 'static {
     /// Chain id.
     fn chain(&self) -> Chain;
 
+    /// Get the [`BaseFeeParams`] for the chain at the given block.
+    fn base_fee_params_at_block(&self, block_number: u64) -> BaseFeeParams;
+
     /// Get the [`BaseFeeParams`] for the chain at the given timestamp.
     fn base_fee_params_at_timestamp(&self, timestamp: u64) -> BaseFeeParams;
```

Comment:
we can keep this here for now, but the only this exists is OP, so maybe we should consider removing that if it doesn't break code that is applicable for both eth and op